state-transition/core/state_processor_genesis.go

```diff
@@ -41,6 +41,15 @@ func (sp *StateProcessor) InitializePreminedBeaconStateFromEth1(
 	execPayloadHeader *ctypes.ExecutionPayloadHeader,
 	genesisVersion common.Version,
 ) (transition.ValidatorUpdates, error) {
+	if !version.Equals(genesisVersion, execPayloadHeader.GetForkVersion()) {
```

Comment:
this check can be removed once GetForkVersion removed from ExecPayloadheader

Reply:
https://github.com/berachain/beacon-kit/pull/2686/commits/ef596d8bb815244b3806025d004b768d1f72ace0